Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs

```diff
@@ -427,15 +430,33 @@ public override ComponentState GetComponentState()
             {
                 drawDepth = BuckledTo.SpriteComponent.DrawDepth - 1;
             }
+            
 
-            return new BuckleComponentState(Buckled, drawDepth);
+            return new BuckleComponentState(Buckled, drawDepth, EntityBuckledTo, DontCollide);
         }
 
         bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
         {
             return TryUnbuckle(eventArgs.User);
         }
 
+
+        public void Update()
+        {
+            if (!DontCollide || Body == null)
+                return;
```

Comment:
Ideally you don't iterate over buckled components if not relevant.

Just raise an event on the eventbus so the system knows when to iterate over it and when not to. I only ask because this will be called for every buckleable mob on the server (i.e. every player and NPC) every tick.

Reply:
Hmm I really don't know how to do that.
By the way I took inspiration from the ClimbingComponent.cs where it's done like this:
```
public void Update()
        {
            if (!IsClimbing || Body == null)
                return;

            if (_climbController != null && (_climbController.IsBlocked || !_climbController.IsActive))
            {
                if (Body.TryRemoveController<ClimbController>())
                {
                    _climbController = null;
                }
            }

            if (IsClimbing)
                Body.WakeBody();

            if (!IsOnClimbableThisFrame && IsClimbing && _climbController == null)
                IsClimbing = false;

            IsOnClimbableThisFrame = false;
        }
```

Reply:
Waking the body also disgusts me there and I don't even think it's necessary given the controller is doing it?

Reply:
I just tried removing it from the ClimbingComponent but the character gets stuck and cannot move anymore. I don't really know how climbing works though, sooo it's just a baseless test.

Reply:
I'll need to suss it in game then but the controller should be doing it (by should I mean "even if it's not currently").

Reply:
Controller definitely sets it so you should be okay to remove it.

Reply:
Just leave that part as is for now and put a TODO on it, but also with that being said please make the event so it's not O(n) for number of players on server (climbingcomponent also needlessly updates so isn't the best example).

Once that's done I'm happy to merge it.

Reply:
Sadly I really don't know where to start with this event thing :(